sentry-ruby/lib/sentry/baggage.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'cgi'
+
+module Sentry
+  # A {https://www.w3.org/TR/baggage W3C Baggage Header} implementation.
+  class Baggage
+    SENTRY_PREFIX = 'sentry-'.freeze
+    SENTRY_PREFIX_REGEX = /^sentry-/.freeze
+
+    DSC_KEYS = %w(
+      trace_id
+      public_key
+      sample_rate
+      release
+      environment
+      transaction
+      user_id
+      user_segment
+    ).freeze
+
+    # @return [Boolean]
+    attr_reader :mutable
```

Comment:
it will come up in the future, we are undergoing some architecture decisions about when to freeze the DSC, but it's good to have this class prepared for future iteration.
See https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#freezing-dynamic-sampling-context for some context.

Reply:
and it will also be used in the subsequent PR which will create new baggage entries when the ruby SDK is a 'head SDK' (i.e. the one which originates a trace)

Reply:
Thanks for the info 👍